src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,132 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationValidatorException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int AGE_LESS_MIN = 17;
+    private static final int AGE_LESS_ZERO = -10;
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+    private User testUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void init() {
+        testUser = new User();
+        testUser.setLogin("nicolas");
+        testUser.setPassword("123456");
+        testUser.setAge(20);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(RegistrationValidatorException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        testUser.setLogin(null);
+
+        assertThrows(RegistrationValidatorException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_existingUserInStorage_notOk() {
+        storageDao.add(testUser);
+
+        assertThrows(RegistrationValidatorException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_addNewUserInStorage_ok() {
+        testUser.setLogin("bob");
+        User expectedUser = testUser;
+        User actualUser = registrationService.register(testUser);
+
+        assertEquals(expectedUser, actualUser, "User has not been added to the storage");
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        testUser.setAge(null);
+
+        assertThrows(RegistrationValidatorException.class, () -> {
+            registrationService.register(testUser);
+        });
+    }
+
+    @Test
+    void register_ageLessThan18_notOk() {
```

Comment:
dont put concrete values into method names, those required values can be changed in the future

Reply:
not fixed